src/geode/io/model/msh_input.cpp

```diff
@@ -1083,13 +1108,52 @@ namespace
             {
                 filter_duplicated_surface_vertices( s, brep_ );
                 auto surface_builder = builder_.surface_mesh_builder( s.id() );
-                for( const auto v : geode::Range{ s.mesh().nb_vertices() } )
+                const auto& mesh = s.mesh();
+                for( const auto v : geode::Range{ mesh.nb_vertices() } )
                 {
                     surface_builder->set_point(
                         v, nodes_[brep_.unique_vertex(
                                { s.component_id(), v } )] );
                 }
                 surface_builder->compute_polygon_adjacencies();
+                std::vector< geode::PolygonEdge > polygon_edges;
+                for( const auto& line : brep_.internal_lines( s ) )
+                {
+                    const auto& edges = line.mesh();
+                    for( const auto e : geode::Range{ edges.nb_edges() } )
+                    {
+                        const auto e0 = edges.edge_vertex( { e, 0 } );
+                        const auto e1 = edges.edge_vertex( { e, 1 } );
+                        const auto surface0 = brep_.mesh_component_vertices(
+                            brep_.unique_vertex( { line.component_id(), e0 } ),
+                            s.id() );
+                        const auto surface1 = brep_.mesh_component_vertices(
+                            brep_.unique_vertex( { line.component_id(), e1 } ),
+                            s.id() );
+                        for( const auto v0 : surface0 )
+                        {
+                            for( const auto v1 : surface1 )
+                            {
+                                if( const auto edge0 =
+                                        mesh.polygon_edge_from_vertices(
+                                            v0, v1 ) )
+                                {
+                                    polygon_edges.emplace_back( edge0.value() );
+                                }
+                                if( const auto edge1 =
+                                        mesh.polygon_edge_from_vertices(
+                                            v1, v0 ) )
+                                {
+                                    polygon_edges.emplace_back( edge1.value() );
+                                }
+                            }
+                        }
+                    }
+                }
+                for( const auto& edge : polygon_edges )
+                {
+                    surface_builder->unset_polygon_adjacent( edge );
```

Comment:
Why not doing this directly at lines 1141 and 1147 ?

Reply:
Because the polygon_edge_from_vertices is using polygon_adjacent_edge which is failing since the adjacency symmetry is temporary wrong.